### PR TITLE
feat: Add Global Setting Support for Auto Hide Feature

### PR DESCRIPTION
**Pull request by SauceX22**
## **Description:**
Hi, I've enjoyed this extension for a while now, thank you for making this, but one thing that always annoyed me was that the settings were tied to the workspace/project, which isn't good especially for team based projects were people might accidentally push their config of this extension and have conflicts with other team members, also there isn't a reason to have this tied to a workspace/project as this is a User Preference IMO. 

Either way, this PR enhances the existing `vscode-autohide` extension by introducing support for managing auto-hide settings globally, in addition to the existing workspace settings. Users can now toggle the visibility of the side bar, auxiliary bar, and panel globally across all workspaces allowing for consistent behavior across all projects, in addition to being able to override the settings as you used to through workspace settings (aka `.vscode/settings.json`).

Code_kv9JRnt7X1

## **Key Changes:**
- **New Commands:** Added a set of global commands to toggle the auto-hide settings independently of the workspace settings.
  - `autoHide.toggleHideSideBarGlobal`
  - `autoHide.toggleHideAuxiliaryBarGlobal`
  - `autoHide.toggleHidePanelGlobal`
- **Command Organization:** Retained the original workspace-based behavior for overriding the global setting.
  - `autoHide.toggleHideSideBarWorkspace`
  - `autoHide.toggleHideAuxiliaryBarWorkspace`
  - `autoHide.toggleHidePanelWorkspace`

## **Testing:**
- Tested the new global commands in the Extension Development Host to ensure they correctly update the global configuration.
- Verified that workspace commands continue to function as expected, overriding global settings when configured.

## **Why This Change?**
This improvement was made to provide more flexibility for users who prefer consistent behavior across multiple projects. It reduces the need to configure the extension in every workspace, making it more convenient and efficient to use. In addition, to remove the burden of having to exclude that single line from every git commit to avoid messing up the workspace settings for other people if they so happen to use the same extension.

## **Feedback:**
Please review the changes and let me know if there are any adjustments or improvements you'd like to see. I'm open to feedback and ready to make necessary revisions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,19 +75,34 @@
         },
         "commands": [
             {
-                "command": "autoHide.toggleHideSideBar",
+                "command": "autoHide.toggleHideSideBarWorkspace",
```
Comment (sirmspencer) [inline package.json]: Leave all setting and commands the same, other wise its a breaking change.

**Comment by sirmspencer** [comment]
This doesnt follow the pattern in vscode.  You dont need separate commands.  Read in the global settings in the existing commands and merge with the workspace settings.

**Review by sirmspencer — CHANGES_REQUESTED** [review]
Good catch.  I didnt write this part or use it so I didnt notice.  

This change doesnt follow the pattern in vscode. You dont need separate commands. Read in the global settings in the existing commands and merge with the workspace settings.

File: package.json (hunk 1 above)
Comment (SauceX22) [reply package.json]: Workspace commands now reverted back 👍

**Comment by SauceX22** [comment]
> Good catch. I didnt write this part or use it so I didnt notice.
> 
> This change doesnt follow the pattern in vscode. You dont need separate commands. Read in the global settings in the existing commands and merge with the workspace settings.

Hi, that's right, I don't NEED separate commands to have the setting on a global basis, i could set the property manually in my user settings, but i would like a quicker way to toggle this functionality through the command palette instead of going through settings each time as i do toggle between them pretty often per session depending on the task at hand, it's a quality of life feature more than anything tbh, and it would be nice to have

**Comment by sirmspencer** [comment]
> > Good catch. I didnt write this part or use it so I didnt notice.
> > This change doesnt follow the pattern in vscode. You dont need separate commands. Read in the global settings in the existing commands and merge with the workspace settings.
> 
> Hi, that's right, I don't NEED separate commands to have the setting on a global basis, i could set the property manually in my user settings, but i would like a quicker way to toggle this functionality through the command palette instead of going through settings each time as i do toggle between them pretty often per session depending on the task at hand, it's a quality of life feature more than anything tbh, and it would be nice to have

I dont agree.   Pretty much every other setting and such is assumed that you can have global settings, and over write them if you add workspace settings.  I dont want to add global / workspace specific commans.  Just think of all the confusion if you cant remember if "this" repo is using global or workspace settings.  "no seperate command" means the existing command should check if you have workspace settings or global settings and then check which one to update.

File: package.json [hunk 2]
```diff
@@ -105,4 +120,4 @@
         "typescript": "^2.9.2",
         "vscode": "^1.1.36"
     }
-}
+}
```
Comment (sirmspencer) [inline package.json]: single blank line at the end of very file.

Comment (SauceX22) [reply package.json]: added back the blank line and now its good, this is the default behavior of vscode formatter

**Comment by SauceX22** [comment]
I have fully reverted the "for Current Workspace" part of the commands and made the rest match the style

**Comment by SauceX22** [comment]
whoops i realized i removed the blank line again, should be good now

**Comment by SauceX22** [comment]
> > > Good catch. I didnt write this part or use it so I didnt notice.
> > > This change doesnt follow the pattern in vscode. You dont need separate commands. Read in the global settings in the existing commands and merge with the workspace settings.
> > 
> > 
> > Hi, that's right, I don't NEED separate commands to have the setting on a global basis, i could set the property manually in my user settings, but i would like a quicker way to toggle this functionality through the command palette instead of going through settings each time as i do toggle between them pretty often per session depending on the task at hand, it's a quality of life feature more than anything tbh, and it would be nice to have
> 
> I dont agree. Pretty much every other setting and such is assumed that you can have global settings, and over write them if you add workspace settings. I dont want to add global / workspace specific commans. Just think of all the confusion if you cant remember if "this" repo is using global or workspace settings. "no seperate command" means the existing command should check if you have workspace settings or global settings and then check which one to update.

brother, i realize what i wrote sounded that it's a personal preference to have a global command, and that is NOT what i meant. 

secondly, your opinion is correct and i fully agree, i wish there was no global/workspace specific commands, but the reality is that the default behavior of the extension SHOULD HAVE BEEN that this is a user preference setting, as in, "this is how i use my editor" not "how edit this repo", just like font size and type, just like the size of the panel, thus one would assume 

you're talking about the confusion, ofcourse there's going to be confusion, but this should've been the default behavior of the extension from the begining, change the global value unless spefically told to change the workspace setting manually (or with another cmd if you'd like), that's how one would assume the extension to work, instead this is the situation we're in where the default behavior changes the workspace setting no matter what, and not to break backwards compat, you have to add another separate command

**Comment by sirmspencer** [comment]
The readme includes full instructions on how to publish.  Feel free to fork and publish your own version.  just please keep the name unique.

**Pull request closed without merging**